packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/PostProcessing/PostProcessor.cs

```diff
@@ -10,23 +10,22 @@
 using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Microsoft.CodeAnalysis.Simplification;
+using Microsoft.Generator.CSharp.Providers;
 
 namespace Microsoft.Generator.CSharp
 {
-    internal class PostProcessor
+    public class PostProcessor
     {
         private readonly string? _modelFactoryFullName;
         private readonly string? _aspExtensionClassName;
         private readonly HashSet<string> _typesToKeep;
         private INamedTypeSymbol? _modelFactorySymbol;
 
-        public PostProcessor(
-            HashSet<string> typesToKeep,
-            string? modelFactoryFullName = null,
-            string? aspExtensionClassName = null)
+        public PostProcessor(string? aspExtensionClassName = null)
         {
-            _typesToKeep = typesToKeep;
-            _modelFactoryFullName = modelFactoryFullName;
+            _typesToKeep = [..CodeModelPlugin.Instance.TypeFactory.UnionTypes, .. CodeModelPlugin.Instance.TypesToKeep];
```

Comment:
I know this was previously written at where this class is called, but why we put `unionTypes` inside here?
`UnionTypes` in `TypeFactory represents the set of names of all existing union types if they have a name.
There is no reason that we need to put them as "types to keep"
@JoshLove-msft do you know anything about this?
Only tracking the name is also an anti-pattern. Majority of unions do not have names, therefore majority of them should be empty string.
anyhow, I opened an issue for this: https://github.com/microsoft/typespec/issues/5673
We could reconsider whether we should keep it but we should remove its reference here.